### ISSUE TITLE
Inconsistencies with new chunking heuristics

### ISSUE THREAD

**Issue by kmuehlbauer** [h5netcdf/h5netcdf#149]
#### What happened:

While trying to adapt to the changes here over at xarray I found some inconsistencies in the behavior.

So I was assuming that the unlimited dimension is always created with chunksize of 1 in the below circumstances.

#### What you expected to happen:

`h5netcdf` heuristics should give chunksize of `1` in the below example.

#### MCVE Code Sample

```python
import h5netcdf
import numpy as np
tmp_local_netcdf = "test.h5"
print("wanted (1, 4, 5, 10)")

print("\nactual first dimension size: 1, without data")
with h5netcdf.File(tmp_local_netcdf, "w") as ds:
    ds.dimensions = {"time": None, "x": 4, "y": 5, "z": 10}
    ds.resize_dimension("time", 1)
    v = ds.create_variable("hello0", ("time", "x", "y", "z"), "float", chunks="h5netcdf")
    print("h5netcdf", v.chunks)
    v = ds.create_variable("hello1", ("time", "x", "y", "z"), "float", chunks="h5py")
    print("h5py ", v.chunks)

print("\nactual first dimension size: 1, with data (1, 4, 5, 10)")
with h5netcdf.File(tmp_local_netcdf, "w") as ds:
    ds.dimensions = {"time": None, "x": 4, "y": 5, "z": 10}
    ds.resize_dimension("time", 1)
    data = np.ones((1, 4, 5, 10))
    v = ds.create_variable("hello0", ("time", "x", "y", "z"), "float", data=data, chunks="h5netcdf")
    print("h5netcdf", v.chunks)
    v = ds.create_variable("hello1", ("time", "x", "y", "z"), "float", data=data, chunks="h5py")
    print("h5py", v.chunks)

print("\nactual first dimension size: 0, without data")
with h5netcdf.File(tmp_local_netcdf, "w") as ds:
    ds.dimensions = {"time": None, "x": 4, "y": 5, "z": 10}
    v = ds.create_variable("hello0", ("time", "x", "y", "z"), "float", chunks="h5netcdf")
    print("h5netcdf", v.chunks)
    v = ds.create_variable("hello1", ("time", "x", "y", "z"), "float", chunks="h5py")
    print("h5py", v.chunks)
```

```
wanted (1, 4, 5, 10)

actual first dimension size: 1, without data
h5netcdf (4, 4, 5, 10)
h5py  (1, 4, 5, 10)

actual first dimension size: 1, with data (1, 4, 5, 10)
h5netcdf (4, 4, 5, 10)
h5py (1, 4, 5, 10)

actual first dimension size: 0, without data
h5netcdf (4, 4, 5, 10)
h5py (256, 1, 2, 5)


```

#### Expected Output

wanted (1, 4, 5, 10)

actual first dimension size: 1, without data
h5netcdf (1, 4, 5, 10)
h5py  (1, 4, 5, 10)

actual first dimension size: 1, with data (1, 4, 5, 10)
h5netcdf (1, 4, 5, 10)
h5py (1, 4, 5, 10)

actual first dimension size: 0, without data
h5netcdf (1, 4, 5, 10)
h5py (256, 1, 2, 5)

#### Version

latest main-branch

**Comment by kmuehlbauer** [comment]
@dionhaefner Could you please clarify, if this is wanted behavior? Thanks.

**Comment by dionhaefner** [comment]
> So I was assuming that the unlimited dimension is always created with chunksize of 1 in the below circumstances.

The unlimited dimensions only collapse to 1 if the other dimensions are bigger, to avoid excessively small chunks. For example, in the trivial case where `shape=(None,)` you get `chunks=(1024,)`. So this part is working correctly.

> actual first dimension size: 1, with data (1, 4, 5, 10)
> h5netcdf (4, 4, 5, 10)
> h5py (1, 4, 5, 10)

Now this part is inconsistent. It seems like `h5py` uses the *current* shape to determine chunks, not the *maximum* shape. And only if the current shape is unset it uses the default.

I assume that you prefer to stay compliant with `h5py`. In this case let me know and I can throw together a PR.

**Comment by kmuehlbauer** [comment]
Yes, I think this will probably help users to adapt to the new heuristics without too much differences. It would be great if you could also add your suggestion from #147 then.

**Comment by kmuehlbauer** [comment]
Resolved by #150
